src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  protected sidesArray: number[] = [this.a, this.b, this.c];
+
+  protected longestSide: number = this.sidesArray.splice(
+    this.sidesArray.indexOf(Math.max(...this.sidesArray)),
+  )[0];
+
+  protected otherSidesSum: number = this.sidesArray.reduce(
+    (acc: number, sideLength: number) => acc + sideLength,
+    0,
+  );
+
+  protected semiperimeter: number = (this.longestSide + this.otherSidesSum) / 2;
 
-export class Triangle implements Figure {}
+  protected isSidesWrong: boolean = this.longestSide >= this.otherSidesSum;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    readonly shape: Shape = Shape.triangle,
+  ) {
+    if (this.sidesArray.some((side) => side <= 0)) {
+      throw new Error('length of some side is <= 0');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in the error message. Consider rephrasing the error message to be more descriptive, such as 'Triangle side length must be greater than 0'.